src/NetMQ/NetMQSocket.cs

```diff
@@ -343,18 +343,10 @@ internal void InvokeEvents(object sender, PollEvents events)
             m_socketEventArgs.Init(events);
 
             if (events.HasIn())
-            {
-                var temp = m_receiveReady;
-                if (temp != null)
-                    temp(sender, m_socketEventArgs);
-            }
+                m_receiveReady?.Invoke(sender, m_socketEventArgs);
```

Comment:
I'm not sure how null propagation is working, but this might be not safe if null propagation is not copying the variable before the null check. the temp variable is used to avoid the race condition.